Libpostal test fix

xGabeC opened the pull request:
Updates test_address_equivalence.rb tests for new Addressor update to libpostal

xGaben committed:
Created docker environment for testing with ruby & libpostal

xGaben committed:
Updated old address_equivalence tests for libpostal update

chanson02 commented on test/test_address_equivalence.rb:
```diff
@@ -13,255 +13,246 @@ def test_different_street_number
+  end
+
+  def test_country
+    assert_same_full(
```
This will fail now, we need to use the `countryless` hash instead of full.

xGabeC replied:
Added a assert_same_countryless

chanson02 commented on test/test_address_equivalence.rb:
```diff
@@ -13,255 +13,246 @@ def test_different_street_number
+
+  
+  def test_direction_as_street
+    assert !NYAddressor.process('901 Avenue E, Wisner, NE 68791, United States').fingerprints.nil?
```
Will this ever fail? If the address fails to parse would it just make
```ruby
fingerprints = { full: nil, unitless: nil, ... }
```

xGabeC replied:
I will update this to check if the full hash is nil. I originally was testing with a string that caused an error, so it was still failing regardless. I didnt realize that it would return like that.

xGaben committed:
Updated from PR Comments: https://github.com/nearestyou/ny-addressor/pull/4

xGaben committed:
Updated tests from feedback: https://github.com/nearestyou/ny-addressor/pull/5

chanson02 commented on test/test_helper.rb:
```diff
@@ -53,6 +53,12 @@ def assert_same_zipless(a, b, msg=nil)
   end
 
+  def assert_same_countryless(a, b, msg=nil)
+    assert_with_diag("zipless fingerprint mismatch", a, b) do
```
```suggestion
    assert_with_diag("countryless fingerprint mismatch", a, b) do
```

chanson02 commented on Dockerfile:
```diff
@@ -0,0 +1,31 @@
+COPY . .
+
+# 📦 Install Ruby gems
+RUN gem install bundler && bundle install
```
```suggestion
# 📦 Install Ruby gems
COPY Gemfile Gemfile.lock ./
COPY ny-addressor.gemspec ./
COPY lib/ny-addressor/version.rb lib/ny-addressor/version.rb
RUN gem install bundler && bundle install

COPY . .
```

xGabeC committed:
Update debug message for assert_same_countryless test

Co-authored-by: Cooper <66448807+chanson02@users.noreply.github.com>

xGabeC committed:
Update Dockerfile

Switch full copy to after bundle install, so docker can better cache the dependancies

Co-authored-by: Cooper <66448807+chanson02@users.noreply.github.com>
